### PR TITLE
Move from Select2 to SelectWoo

### PR DESCRIPTION
**Pull request by bordoni**
🎟 [TCMN-23]
--- 

There are 4 Pull Requests that go along with this one for all of our products that have used for Select2, we are migrating them to SelectWoo.

_Related PRs:_
* https://github.com/moderntribe/events-community/pull/467
* https://github.com/moderntribe/the-events-calendar/pull/3119
* https://github.com/moderntribe/tribe-common/pull/1327
* https://github.com/moderntribe/events-pro/pull/1524
* https://github.com/moderntribe/events-filterbar/pull/252

[TCMN-23]: https://moderntribe.atlassian.net/browse/TCMN-23

### CODE REVIEW

File: readme.txt [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 == Changelog ==
 
+= [4.11.6] TBD =
+
```
Comment (juanfra) [inline readme.txt:6]: We'll need to set something here if we're merging to master.

File: src/resources/js/dependency.js [hunk 2]
```diff
@@ -160,12 +160,13 @@
 			}
 
 			if ( passes && ! isDisabled ) {
-				if ( $dependent.data( 'select2' ) ) {
-					$dependent.data( 'select2' ).container.addClass( activeClass );
+				console.log( $dependent );
```
Comment (juanfra) [inline src/resources/js/dependency.js:163]: Let's remove the console log

File: src/resources/js/dropdowns.js [hunk 3]
```diff
@@ -220,22 +241,22 @@ var tribe_dropdowns = tribe_dropdowns || {};
...
+				delete args.containerCss;
 			}
 		}
 
-		// How do we match the Search
-
-		// TODO: Revisit
-		//args.matcher = obj.matcher;
+		// How do we match the Search.
+		// args.matcher = obj.matcher;
```
Comment (sc0ttkclark) [inline src/resources/js/dropdowns.js:256]: Keep?

File: src/resources/js/dropdowns.js [hunk 4]
```diff
@@ -254,11 +275,12 @@ var tribe_dropdowns = tribe_dropdowns || {};
 		if ( $select.is( '[data-options]' ) ) {
 			args.data = $select.data( 'options' );
 
-			if ( ! $select.is( 'select' ) ) {
-				args.initSelection = obj.init_selection;
-			}
+			// if ( ! $select.is( 'select' ) ) {
```
Comment (sc0ttkclark) [inline src/resources/js/dropdowns.js:278]: Keep?

File: src/resources/js/dropdowns.js [hunk 5]
```diff
@@ -321,7 +344,7 @@ var tribe_dropdowns = tribe_dropdowns || {};
 		if ( $select.is( '[data-tags]' ) ) {
 			args.tags = $select.data( 'tags' );
 
-			args.initSelection = obj.init_selection;
+			// args.initSelection = obj.init_selection;
```
Comment (sc0ttkclark) [inline src/resources/js/dropdowns.js:347]: Keep?

File: src/resources/js/dropdowns.js [hunk 6]
```diff
@@ -400,33 +426,30 @@ var tribe_dropdowns = tribe_dropdowns || {};
...
 		// Propagating original input classes to the select2 container.
-		$container.data( 'select2' ).$container.addClass( args.originalClasses );
+		$container.data( 'select2' ).$container.addClass( obj.getSelectClasses( $select ).join( ' ' ) );
 
 		// Propagating original input classes to the select2 container.
 		$container.data( 'select2' ).$container.removeClass( 'hide-before-select2-init' );
 
-		// TODO: Revisit
-		//
 		// if ( carryOverData.length > 0 ) {
```
Comment (sc0ttkclark) [inline src/resources/js/dropdowns.js:435]: Keep?

File: readme.txt (hunk 1 above)
Comment (bordoni) [reply readme.txt:6]: Still TBD on what will actually be the Changelog. Will update once we determine  that, we will run with an In Branch QA

File: src/resources/js/dropdowns.js [hunk 7]
```diff
@@ -172,45 +209,52 @@ var tribe_dropdowns = tribe_dropdowns || {};
...
+			'condition-is-checked',
 		];
+
 		var $container;
 
 		// Add a class for dropdown created
 		$select.addClass( obj.selector.created.className() );
 
-		// For Reference we save the jQuery element as an Arg
+		args.debug = true;
```
Comment (paulmskim) [inline src/resources/js/dropdowns.js:220]: is this for debugging purposes? do you need to remove this?

File: src/resources/js/dropdowns.js [hunk 8]
```diff
@@ -172,45 +209,52 @@ var tribe_dropdowns = tribe_dropdowns || {};
...
 
 		// CSS for the dropdown
 		args.dropdownCss = {};
 		args.dropdownCss.width = 'auto';
 
-		// When we have this we replace the default with what's in the param
+		// When we have this we replace the default with what's in the param.
 		if ( $select.is( '[data-dropdown-css-width]' ) ) {
 			args.dropdownCss.width = $select.data( 'dropdown-css-width' );
+			console.log( args.dropdownCss.width );
```
Comment (paulmskim) [inline src/resources/js/dropdowns.js:248]: console log

File: src/resources/js/dropdowns.js [hunk 9]
```diff
@@ -359,20 +411,29 @@ var tribe_dropdowns = tribe_dropdowns || {};
...
+		$container = $select.select2( args );
+
+		// Propagating original input classes to the select2 container.
+		$container.data( 'select2' ).$container.addClass( obj.getSelectClasses( $select ).join( ' ' ) );
 
+		// Propagating original input classes to the select2 container.
+		$container.data( 'select2' ).$container.removeClass( 'hide-before-select2-init' );
+
+		/**
+		 * @todo @bordoni Investigate how and if we should be doing this.
```
Comment (paulmskim) [inline src/resources/js/dropdowns.js:431]: still working on this?

File: src/resources/js/dropdowns.js [hunk 10]
```diff
@@ -381,11 +442,13 @@ var tribe_dropdowns = tribe_dropdowns || {};
...
 			}, $container );
 		}
+		 */
 	};
 
-	obj.action_change =  function( event ) {
-		var $select = $( this ),
-			data = $( this ).data( 'value' );
+	obj.action_change = function( event ) {
+		return;
```
Comment (paulmskim) [inline src/resources/js/dropdowns.js:449]: is this on purpose? do we need this function if you're not doing anything here?

File: src/resources/js/dropdowns.js (hunk 9 above)
Comment (bordoni) [reply src/resources/js/dropdowns.js:425]: It's ongoing until QA finishes their pass. Hopefully, I can clear that todo before merging.

File: src/resources/js/dropdowns.js [hunk 11]
```diff
@@ -269,6 +269,11 @@ var tribe_dropdowns = window.tribe_dropdowns || {};
 			args.escapeMarkup = obj.allow_html_markup;
 		}
 
+		// If we are dealing with a Input Hidden we need to set the Data for it to work
```
Comment (sc0ttkclark) [inline src/resources/js/dropdowns.js:272]:
```suggestion
		// If we are dealing with a Input Hidden we need to set the Data for it to work.
```

File: src/resources/js/dropdowns.js [hunk 12]
```diff
@@ -226,27 +260,38 @@ var tribe_dropdowns = tribe_dropdowns || {};
...
+		if ( ! $select.is( '[data-placeholder]' ) && $select.is( '[placeholder]' ) ) {
+			args.placeholder = $select.attr( 'placeholder' );
 		}
 
 		// Don't Remove HTML elements or escape elements
 		if ( $select.is( '[data-allow-html]' ) ) {
 			args.escapeMarkup = obj.allow_html_markup;
 		}
 
+		// If we are dealing with a Input Hidden we need to set the Data for it to work
```
Comment (sc0ttkclark) [inline src/resources/js/dropdowns.js:272]:
```suggestion
		// If we are dealing with a Input Hidden we need to set the Data for it to work.
```

**Pull request merged**